app/models/katello/host/subscription_facet.rb

```diff
@@ -34,6 +35,19 @@ def import_database_attributes(consumer_params)
         end
       end
 
+      def update_consumer_type(consumer_params)
+        if consumer_params.try(:[], 'type').try(:[], 'label') == 'hypervisor'
+          self.consumer_type = 'hypervisor'
+        elsif consumer_params.try(:[], 'facts').try(:[], 'virt.is_guest') == "true"
+          self.consumer_type = 'guest'
+        elsif !consumer_params.try(:[], 'guestIds').empty?
```

Comment:
We do utilize a fact parser here:  https://github.com/Katello/katello/blob/master/app/models/katello/rhsm_fact_parser.rb and likely could use that parser to do this functionality. 

So far we've only utilized the FactParser for attributes that the normal fact import process handles (such as operatingsystem, interfaces, etc..).  But no reason we couldn't use it for other things too.